llvm/lib/Target/AMDGPU/AMDGPURegBankLegalizeHelper.cpp

```diff
@@ -127,6 +131,120 @@ void RegBankLegalizeHelper::widenLoad(MachineInstr &MI, LLT WideTy,
   MI.eraseFromParent();
 }
 
+bool isSignedBFE(MachineInstr &MI) {
+  if (isa<GIntrinsic>(MI)) {
+    switch (MI.getOperand(1).getIntrinsicID()) {
+    case Intrinsic::amdgcn_sbfe:
+      return true;
+    case Intrinsic::amdgcn_ubfe:
+      return false;
+    default:
+      llvm_unreachable("Opcode not supported");
+    }
+  } else {
+    switch (MI.getOpcode()) {
+    case AMDGPU::G_SBFX:
+      return true;
+    case AMDGPU::G_UBFX:
+      return false;
+    default:
+      llvm_unreachable("Opcode not supported");
+    }
+  }
+}
+
+void RegBankLegalizeHelper::lowerDiv_BFE(MachineInstr &MI) {
```

Comment:
"Div" may be confused with division. This also isn't really about divergence, but VGPR. lowerV_BFE?